### PR TITLE
[dom] WIP Easyconfig for Ipython-5.4.1 for Craygnu-17.08, Python2 and dependencies 

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Comment by teojgo** [comment]
retest this please

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 1]
```diff
@@ -0,0 +1,224 @@
+# Contributed by Theofilos Manitaras (CSCS)
+easyblock = 'Bundle'
+
+name = 'IPython'
+version = '5.4.1'
+versionsuffix = '-Python-2.7.13'
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: `-python2`

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+homepage = 'http://ipython.org/index.html'
+description = """IPython provides a rich architecture for interactive computing with:
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: reformat:

```
description = """IPython provides a rich architecture for interactive computing
with: Powerful interactive shells (terminal and Qt-based).  A browser-based
notebook with support for code, text, mathematical expressions, inline plots
and other rich media. Support for interactive data visualization and use of GUI
toolkits. Flexible, embeddable interpreters to load into your own projects.
Easy to use, high performance tools for parallel computing."""
```

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+ A browser-based notebook with support for code, text, mathematical expressions, inline plots and other rich media.
+ Support for interactive data visualization and use of GUI toolkits.
+ Flexible, embeddable interpreters to load into your own projects.
+ Easy to use, high performance tools for parallel computing."""
+
+toolchain = {'name': 'CrayGNU', 'version': '17.08'}
+
+dependencies = [
+    ('cray-python/17.06.1', EXTERNAL_MODULE),
+    ('PyZMQ', '17.0.0b1', '%s-zmq4' % versionsuffix),
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: :warning: versionsuffix

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+]
+
+pyshortver='2.7'
+
+toolchainopts = {'pic': True, 'verbose' : False }
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: move up to line 16

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb (hunk 1 above)
Comment (jgphpc) [reply easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: rename the .eb too

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        'source_urls': ['https://pypi.python.org/packages/source/i/ipython/'],
+        'modulename': 'IPython',
+    }),
+    ('ipykernel', '4.6.1', {
+        'source_urls': ['https://pypi.python.org/packages/0c/41/67e16b243b78'  \
+                        'b49f4b1650d045b63be187c27d20a76f0f7b8e61e0fcb966/'],
+    }),
+]
+
+modextrapaths = {'PYTHONPATH': ['lib/python2.7/site-packages']}
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: use `%(pyshortver)s` ?

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+modextravars = {'LC_ALL' : 'en_US.UTF-8',
+                'LC_CTYPE' : 'en_US.UTF-8',
+                'LANG' : 'en_US.UTF-8'}
+
+sanity_check_paths = {
+    'files': ['bin/ipython'],
+    'dirs': ['lib/python2.7/site-packages/IPython'],
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: `python2.7` => pyshortver

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb (hunk 1 above)
Comment (jgphpc) [reply easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb]: ok i see it now. `git rm IPython-5.4.1-CrayGNU-17.08-Python-2.7.13.eb`

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb [hunk 7]
```diff
@@ -0,0 +1,231 @@
+# Contributed by Theofilos Manitaras (CSCS)
+easyblock = 'Bundle'
+
+name = 'IPython'
+version = '5.4.1'
+versionsuffix = '-cray-python2'
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb]: -python2

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+homepage = 'http://ipython.org/index.html'
+description = """IPython provides a rich architecture for interactive computing with:
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb]: format...

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+        'source_urls': ['https://pypi.python.org/packages/source/i/ipython/'],
+        'modulename': 'IPython',
+    }),
+    ('ipykernel', '4.6.1', {
+        'source_urls': ['https://pypi.python.org/packages/0c/41/67e16b243b78'  \
+                        'b49f4b1650d045b63be187c27d20a76f0f7b8e61e0fcb966/'],
+    }),
+]
+
+modextrapaths = {'PYTHONPATH': ['lib/python2.7/site-packages']}
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb]: pyshortver

File: easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+
+modextravars = {'LC_ALL' : 'en_US.UTF-8',
+                'LC_CTYPE' : 'en_US.UTF-8',
+                'LANG' : 'en_US.UTF-8'}
+
+sanity_check_paths = {
+    'files': ['bin/ipython'],
+    'dirs': ['lib/python2.7/site-packages/IPython'],
```
Comment (jgphpc) [inline easybuild/easyconfigs/i/IPython/IPython-5.4.1-CrayGNU-17.08-cray-python2.eb]: pyshortver

File: easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-Python-2.7.13-zmq4.eb [hunk 11]
```diff
@@ -0,0 +1,39 @@
+# Contributed by Theofilos Manitaras (CSCS)
```
Comment (jgphpc) [inline easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-Python-2.7.13-zmq4.eb]: `git rm`

File: easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-Python-2.7.13-zmq4.eb [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+python = 'Python'
+python_version = '2.7.13'
+zmq_version = '4.2.2'
+zmq_major_version = zmq_version.split('.')[0] 
+python_short_version = '2.7' 
+
+versionsuffix = '-Python-2.7.13-zmq{0}'.format(zmq_major_version)
+
+configopts = '--zmq=~/easybuild/dom/haswell/software/ZeroMQ/4.2.2-CrayGNU-17.08/'
```
Comment (jgphpc) [inline easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-Python-2.7.13-zmq4.eb]: ?

File: easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-cray-python2-zmq4.eb [hunk 13]
```diff
@@ -0,0 +1,37 @@
+# Contributed by Theofilos Manitaras (CSCS)
```
Comment (jgphpc) [inline easybuild/easyconfigs/p/PyZMQ/PyZMQ-17.0.0b1-CrayGNU-17.08-cray-python2-zmq4.eb]: `git rm PyZMQ-17.0.0b1-CrayGNU-17.08-Python-2.7.13-zmq4.eb`

**Comment by teojgo** [comment]
Please @jgphpc close this pull request, I created one with the newest version while taking into consideration your comments in this one.

**Pull request closed without merging**